src/shogun2-util/src/main/java/de/terrestris/shogun2/util/interceptor/OgcMessageDistributor.java

```diff
@@ -0,0 +1,370 @@
+
+	/**
+	 *
+	 * @param request
+	 * @param message
+	 * @return
+	 * @throws InterceptorException
+	 */
+	public MutableHttpServletRequest distributeToRequestInterceptor(
+			MutableHttpServletRequest request, OgcMessage message)
+					throws InterceptorException {
+
+		if (message.isRequestAllowed()) {
+			LOG.debug("Request is ALLOWED, not intercepting the request.");
+			return request;
+		} else if (message.isRequestDenied()) {
+			throw new InterceptorException("Request is DENIED, blocking the request.");
+		} else if (message.isRequestModified()) {
+			LOG.debug("Request is to be MODIFIED, intercepting the request.");
+		}
+
+		String implErrMsg = MessageFormat.format(REQUEST_IMPLEMENTATION_NOT_FOUND_MSG,
+				message.getService(), message.getOperation());
+		String infoMsg = MessageFormat.format(MODIFYING_REQUEST_MSG,
+				message.getService(), message.getOperation());
+		String serviceErrMsg = MessageFormat.format(REQUEST_NOT_SUPPORTED_MSG,
+				message.getService());
+		String operationErrMsg = MessageFormat.format(REQUEST_NOT_SUPPORTED_MSG,
+				message.getOperation());
+
+		if (message.isWms()) {
+
+			// check if the wmsRequestInterceptor is available
+			if (this.wmsRequestInterceptor == null) {
+				LOG.debug(implErrMsg);
+				return request;
+			}
+
+			LOG.debug(infoMsg);
+
+			if (message.isWmsGetCapabilities()) {
+				request = this.wmsRequestInterceptor.interceptGetCapabilities(request);
+			} else if (message.isWmsGetMap()) {
+				request = this.wmsRequestInterceptor.interceptGetMap(request);
+			} else if (message.isWmsGetFeatureInfo()) {
+				request = this.wmsRequestInterceptor.interceptGetFeatureInfo(request);
+			} else if (message.isWmsGetLegendGraphic()) {
+				request = this.wmsRequestInterceptor.interceptGetLegendGraphic(request);
+			} else if (message.isWmsGetStyles()) {
+				request = this.wmsRequestInterceptor.interceptGetStyles(request);
+			} else if (message.isWmsDescribeLayer()) {
+				request = this.wmsRequestInterceptor.interceptDescribeLayer(request);
+			} else {
+				throw new InterceptorException(operationErrMsg);
+			}
+
+		} else if (message.isWfs()) {
+
+			// check if the wfsRequestInterceptor is available
+			if (this.wfsRequestInterceptor == null) {
+				LOG.debug(implErrMsg);
+				return request;
+			}
+
+			LOG.debug(infoMsg);
+
+			if (message.isWfsGetCapabilities()) {
+				request = this.wfsRequestInterceptor.interceptGetCapabilities(request);
+			} else if (message.isWfsGetFeature()) {
+				request = this.wfsRequestInterceptor.interceptGetFeature(request);
+			} else if (message.isWfsDescribeFeatureType()) {
+				request = this.wfsRequestInterceptor.interceptDescribeFeatureType(request);
+			} else if (message.isWfsTransaction()) {
+				request = this.wfsRequestInterceptor.interceptTransaction(request);
+			} else if (message.isWfsLockFeature()) {
+				request = this.wfsRequestInterceptor.interceptLockFeature(request);
+			} else {
```

Comment:
The above catches all `v1.1.0` WFS operations, which is tootally OK. We should just explicitly document this somewhere, right?